modules/card.js

```diff
@@ -1,92 +1,281 @@
-        const differentCardsWithoutImage = _.differenceBy(cardsWithoutImage, cardsWithImage, "name");
-        const uniqCards = _.uniqBy(_.concat(cardsWithImage, differentCardsWithoutImage), "name");
+    // generate the embed card
+    generateEmbed(cards, command, hasEmojiPermission) {
+        return new Promise(resolve => {
+            const card = cards[0];
+
+            // generate embed title and description text
+            let title = card.name + ' ' + card.mana_cost;
+            let description = this.generateDescriptionText(card);
+
+            // are we allowed to use custom emojis? cool, then do so
+            if(hasEmojiPermission) {
+                title = this.renderEmojis(title);
+                description = this.renderEmojis(description);
+            }
+
+            // instantiate embed object
+            const embed = new Discord.RichEmbed({
+                title,
+                description,
+                url: card.scryfall_uri,
+                color: this.getBorderColor(card),
+                thumbnail: {url: card.image_uri}
+            });
+
+            // add pricing, if requested
+            if (command === 'price') {
+                let prices = [];
+                if(card.usd) prices.push('$' + card.usd);
+                if(card.eur) prices.push(card.eur + '€');
+                if(card.tix) prices.push(card.eur + ' Tix');
+                embed.addField('Prices', prices.join(' / '));
+            }
+
+            // add legalities, if requested
+            if (command === 'legal') {
+                const legalFormats = Object.keys(card.legalities).filter(format => card.legalities[format] === 'legal');
+                embed.addField('Legal in', legalFormats.map(_.capitalize).join(', '));
+            }
+
+            // add footer, if needed
+            if(cards.length > 1) {
+                let footer = (cards.length - 1) + ' other hits:\n';
+                footer += cards.slice(1,6).map(cardObj => cardObj.name).join('; ');
+                if (cards.length > 6) footer += '; ...';
+                embed.setFooter(footer);
+            }
+
+            // add rulings loaded from Gatherer, if needed
+            if(card.related_uris.gatherer && (command === "ruling" || command === "rulings")) {
+                rp(card.related_uris.gatherer).then(gatherer => {
+                    embed.setAuthor('Gatherer rulings for');
+                    embed.setDescription(this.parseGathererRulings(gatherer).substr(0,2048));
+                    resolve(embed);
+                });
+            } else {
+                resolve(embed);
+            }
+        });
+    }
 
-        return uniqCards.find(c => c.name.toLowerCase() === cardName) ||
-            uniqCards.find(c => c.name.toLowerCase().startsWith(cardName)) || uniqCards[0];
+    // fetch permissions from Guild to use custom emojis
+    getEmojiPermission(msg) {
+        return new Promise(resolve => {
```

Comment:
If you store the whole guild instead, the cache would be a lot bigger (than storing just the permissions) and the permissions themselves would still have to be loaded, because they are not directly loaded with a Guild and need to be "fetched" along with the member you're looking up. This way, we cache exactly what we need: a per-guild permission array used for checking what the bot is and is not allowed to do on a server.